Libraries/VtkVgSceneGraph/vtkVgGeode.cxx

```diff
@@ -145,15 +145,24 @@ void vtkVgGeode::Update(vtkVgNodeVisitorBase& nodeVisitor)
       {
       // \TODO: We are dealing with only 3d props. We need to find a way for
       // setting a user transform to vtkProp if needed.
-      vtkProp3D* currentProp = static_cast<vtkProp3D*>(this->ActiveDrawables->GetNextProp());
-      if (currentProp)
+      vtkProp3D* currentProp =
+        static_cast<vtkProp3D*>(this->ActiveDrawables->GetNextProp());
+      // Divide through by 3,3 component to be assured the 3,3 component is 1
+      // (and more importantly, greater than 0)
+      double w = this->FinalMatrix->GetElement(3, 3);
+      if (currentProp && w != 0)
```

Comment:
This causes the later declaration to shadow. Consider inlining `w` here instead?

Reply:
These changes surprised me... I want to blame it on git.